server/src/test/java/org/elasticsearch/common/xcontent/support/XContentMapValuesTests.java

```diff
@@ -730,4 +732,175 @@ public void testExtractNestedSources() {
             )
         );
     }
+
+    public void testInsertValue() throws Exception {
+        final BiFunction<Map<String, Object>, String, Object> getMapValue = (map, key) -> {
```

Comment:
Why not using `XContentMapValue#extractValue` for this?

Reply:
Because I wanted to create a function to get a map value where I had more control over exactly how the intermediate structures were traversed. This matters, for example, when checking that the key-value pair was added to the root map, or when checking that certain keys were concatenated during traversal.